File: 2298/2298.py
class Solution:
    def tasks_count_in_the_weekend(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.tasks_count_in_the_weekend(root.left), self.tasks_count_in_the_weekend(root.right))

File: 2298/2298.go
func tasksCountInTheWeekend(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := tasksCountInTheWeekend(root.Left)
	r := tasksCountInTheWeekend(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}